java/code/src/com/suse/manager/utils/SaltKeyUtils.java

```diff
@@ -81,8 +180,7 @@ public boolean deleteSaltKey(User user, String minionId) throws PermissionExcept
             });
         }
         else {
-            LOG.info(String.format("No key found for minionID [%s]", minionId));
-            return false;
+            throw new IllegalArgumentException("No key found for minionID [" + minionId + "]");
```

Comment:
Please check, if this change has consequences at other places where this function is used.
I can see SaltServerActionService using it and I think at this place it was wanted to ignore a possible error.

Reply:
I tried to go more in deep on the code. `deleteSaltKey` was used in `SaltServerActionService` and `MinionsAPI`: using `saltApi.deleteKey` instead of `saltKeyUtils.deleteSaltKey` , we should avoid problem with exception and remove at all `SaltKeyUtils` dependencies from `SaltServerActionService` .

Reply:
revert the changes, in this way everything sholud be safe :)